relays/lib-substrate-relay/src/messages_source.rs

```diff
@@ -366,14 +364,10 @@ where
 		id: TargetHeaderIdOf<MessageLaneAdapter<P>>,
 	) -> Result<Option<Self::BatchTransaction>, SubstrateError> {
 		if let Some(ref target_to_source_headers_relay) = self.target_to_source_headers_relay {
-			if P::SourceBatchCallBuilder::BATCH_CALL_SUPPORTED {
-				let (proved_header, prove_calls) =
-					target_to_source_headers_relay.prove_header(id.0).await?;
-				return Ok(Some(BatchConfirmationTransaction {
-					messages_source: self.clone(),
-					proved_header,
-					prove_calls,
-				}))
+			let (proved_header, prove_calls) =
```

Comment:
This changes logic a bit - you always generate a header proof (which is a heavy operation) and then, if batch calls are not supported, you simply drop it. I'd keep the old behavior, where it is only generated when strictly required. OTOH it isn't that critical - we don't care that much about relay performance. If you'd like to keep it, please at least file an issue to fix it

Reply:
Fixed. Thanks ! I completely missed this.